include/boost/url/decoded_view.hpp

```diff
@@ -0,0 +1,1012 @@
+
+    /** Returns an iterator to the end
+
+        Returns an iterator to one past the last decoded character of the string.
+
+        This character acts as a placeholder, attempting to access it
+        results in undefined behavior.
+
+        @return const_iterator to the character following the last character
+    */
+    BOOST_URL_DECL
+    const_iterator
+    end() const noexcept;
+
+    /// @copydoc end()
+    BOOST_URL_DECL
+    const_iterator
+    cend() const noexcept;
+
+    /** Accesses the first character
+
+        Returns the decoded first character in the string.
+
+        The behavior is undefined if `empty() == true`.
+
+        @return reference to the first character
+     */
+    BOOST_URL_DECL
+    const_reference
+    front() const;
+
+    /** Accesses the last character
+
+        Returns the last decoded character in the string
+
+        The behavior is undefined if `empty() == true`.
+
+        @return reference to the first character
+     */
+    BOOST_URL_DECL
+    const_reference
+    back() const;
+
+    /** Returns a pointer to the first encoded character of a string
+
+        Returns a pointer to the underlying character array.
+
+        The pointer is such that the range `[data(); data() + encoded_size())`
+        is valid and the values in it correspond to the values of the
+        string.
+
+        @note Unlike `std::basic_string::data()` and string literals,
+        `data()` may return a pointer to a buffer that is not
+        null-terminated. Therefore, it is typically a mistake to pass
+        data() to a routine that takes just a const char* and expects
+        a null-terminated string.
+
+        @return A pointer to the underlying character array.
+     */
+    BOOST_CONSTEXPR
+    const char*
+    encoded_data() const
+    {
+        return p_;
+    }
+
+    /** Returns the number of char elements in the string
+
+        @return The number of char elements in the string
+    */
+    BOOST_CONSTEXPR
+    size_type
+    size() const noexcept
+    {
+        return dn_;
+    }
+
+    /// @copydoc size()
+    BOOST_CONSTEXPR
+    size_type
+    length() const noexcept
+    {
+        return dn_;
+    }
+
+    /** Returns the number of encoded char elements in the string
+
+        @return The number of encoded char elements in the string
+    */
+    BOOST_CONSTEXPR
+    size_type
+    encoded_size() const noexcept
+    {
+        return n_;
+    }
+
+    /// @copydoc encoded_size()
+    BOOST_CONSTEXPR
+    size_type
+    encoded_length() const noexcept
+    {
+        return n_;
+    }
+
+    /** Returns the maximum number of characters
+
+        The largest possible number of decoded
+        chars objects that can be referred to by
+        a decoded_view.
+
+        @return Maximum number of decoded characters
+    */
+    BOOST_CONSTEXPR
+    size_type
+    max_size() const noexcept
+    {
+        return size_type( -1 );
+    }
+
+    /** Checks whether the string is empty
```

Comment:
The javadoc brief for functions that return information should start with the word "Return"